dbt_coverage/__init__.py

```diff
@@ -124,40 +179,45 @@ def __post_init__(self):
             self.misses = None
             self.coverage = None
 
-    @staticmethod
-    def from_catalog(catalog: Catalog):
-        cov = catalog.get_coverage()
+    @classmethod
+    def from_catalog(cls, catalog: Catalog, cov_type: CoverageType):
+        cov = catalog.coverage(cov_type)
         return CoverageReport(
-            catalog,
+            cls.Type.CATALOG,
+            None,
             cov[0],
             cov[1],
-            {table.name: CoverageReport.from_table(table) for table in catalog.tables.values()}
+            {table.name: CoverageReport.from_table(table, cov_type)
+             for table in catalog.tables.values()}
         )
 
-    @staticmethod
-    def from_table(table: Table):
-        cov = table.get_coverage()
+    @classmethod
+    def from_table(cls, table: Table, cov_type: CoverageType):
+        cov = table.coverage(cov_type)
         return CoverageReport(
-            table,
+            cls.Type.TABLE,
+            table.name,
             cov[0],
             cov[1],
-            {col.name: CoverageReport.from_column(col) for col in table.columns.values()}
+            {col.name: CoverageReport.from_column(col, cov_type) for col in table.columns.values()}
         )
 
-    @staticmethod
-    def from_column(column: Column):
+    @classmethod
+    def from_column(cls, column: Column, cov_type: CoverageType):
+        cov = column.coverage(cov_type)
         return CoverageReport(
-            column,
-            {''} if column.has_doc() else set(),  # Simulate one or zero "subentities" covered
-            {''},  # Pretend columns has one "subentity"
+            cls.Type.COLUMN,
```

Comment:
@sweco just out of curiosity, any specific reason for using `cls` here rather than `CoverageReport` directly?

Reply:
Not really 😊